a3fe/configuration/engine_config.py

```diff
@@ -0,0 +1,194 @@
+"""Configuration classes for SOMD engine configuration."""
+
+__all__ = [
+    "SomdConfig",
+]
+
+import yaml as _yaml
+import os as _os
+from typing import Dict as _Dict
+
+from pydantic import BaseModel as _BaseModel
+from pydantic import Field as _Field
+from pydantic import ConfigDict as _ConfigDict
+
+
+class SomdConfig(_BaseModel):
+    """
+    Pydantic model for holding SOMD engine configuration.
+    """
+    
+    ### Integrator - ncycles modified as required by a3fe ###
+    nmoves: int = _Field(25000, description="Number of moves per cycle")
+    ncycles: int = _Field(60, description="Number of cycles")
+    timestep: float = _Field(4.0, description="Timestep in femtoseconds")
+    constraint: str = _Field("hbonds", description="Constraint type")
+    hydrogen_mass_factor: float = _Field(
+        3.0, 
+        alias="hydrogen mass repartitioning factor",
+        description="Hydrogen mass repartitioning factor"
+    )
+    integrator: str = _Field("langevinmiddle", description="Integration algorithm")
+    inverse_friction: float = _Field(
+        1.0, 
+        description="Inverse friction in picoseconds",
+        alias="inverse friction"
+    )
+    temperature: float = _Field(25.0, description="Temperature in Celsius")
+    # Thermostatting already handled by langevin integrator
+    thermostat: bool = _Field(False, description="Enable thermostat")
+
+    ### Barostat ###
+    barostat: bool = _Field(True, description="Enable barostat")
+    pressure: float = _Field(1.0, description="Pressure in atm")
+
+    ### Non-Bonded Interactions ###
+    cutoff_type: str = _Field(
+        "PME",
+        alias="cutoff type",
+        description="Type of cutoff to use"
+    )
+    cutoff_distance: float = _Field(
+        10.0,
+        alias="cutoff distance",
+        description="Cutoff distance in angstroms"
+    )
+    
+    ### Trajectory ###
+    buffered_coords_freq: int = _Field(
+        5000,
+        alias="buffered coordinates frequency",
+        description="Frequency of buffered coordinates output"
+    )
+    center_solute: bool = _Field(
+        True,
+        alias="center solute",
+        description="Center solute in box"
+    )
+
+    ### Minimisation ###
+    minimise: bool = _Field(True, description="Perform energy minimisation")
+
+    ### Alchemistry - restraints added by a3fe ###
+    perturbed_residue_number: int = _Field(
+        1,
+        alias="perturbed residue number",
+        description="Residue number to perturb"
+    )
```

Comment:
Add fields to hold - use boresch restraints, default False, turn on receptor-ligand restraints mode, default False, lambda array, also morphfile, topfile, crdfile